.github/actions/pingproxy/action.yml

```diff
@@ -0,0 +1,36 @@
+name: 'Configure and run pingproxy'
+description: 'Configures and runs the pingproxy'
+inputs:
+  tls:
+    description: 'Enable TLS'
+    required: false
+    default: false
+runs:
+  using: "composite"
+  steps:
+    - name: Configure
+      run: |
+        echo "[admin]" >> proxy.toml
+        echo "port = \"9998\"" >> proxy.toml
+        echo "[backend]" >> proxy.toml
+        echo "endpoints = [\"127.0.0.1:12321\"]" >> proxy.toml
+      shell: bash
+    - name: Configure TLS
+      run: |
+          if ${{ inputs.tls }}; then
+          step ca certificate --san=127.0.0.1 --ca-url=127.0.0.1:443 --provisioner-password-file=${HOME}/.step/password localhost proxy.crt proxy.key
+          echo "[tls]" >> proxy.toml
```

Comment:
For readability, can we stick the content of `proxy.toml` into a multi-line string and declare it as an environment variable?

https://docs.github.com/en/actions/learn-github-actions/environment-variables
https://stackoverflow.com/questions/3790454/how-do-i-break-a-string-in-yaml-over-multiple-lines